pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotBrokerRestletResource.java

```diff
@@ -129,6 +133,40 @@
     }
   }
 
+
+  @POST
+  @Path("/brokers/instances/{instanceName}/qps")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Consumes(MediaType.TEXT_PLAIN)
+  @ApiOperation(value = "Enable/disable the query rate limiting for a broker instance", notes = "Enable/disable the query rate limiting for a broker instance")
+  @ApiResponses(value = {@ApiResponse(code = 200, message = "Success"), @ApiResponse(code = 400, message = "Bad Request"), @ApiResponse(code = 404, message = "Instance not found"), @ApiResponse(code = 500, message = "Internal error")})
+  public SuccessResponse toggleQueryRateLimiting(
+      @ApiParam(value = "Broker instance name", required = true, example = "Broker_my.broker.com_30000") @PathParam("instanceName") String brokerInstanceName,
+      @ApiParam(value = "ENABLE|DISABLE", allowableValues = "ENABLE, DISABLE", required = true)  @QueryParam("state") String state) {
+    if (brokerInstanceName == null || !brokerInstanceName.startsWith("Broker_")) {
+      LOGGER.info("{} isn't a valid broker instance name", brokerInstanceName);
+      throw new ControllerApplicationException(LOGGER,
+          String.format("'%s' is not a valid broker instance name.", brokerInstanceName), Response.Status.BAD_REQUEST);
```

Comment:
nit: Create an error string, and then use it in log as well as exeption.
Same in other places as well, will not repeat.

Reply:
Removed the duplicate logging.